catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/TransformVisitor.java

```diff
@@ -177,6 +303,21 @@ public void visitPropertyIsNilType(VisitableElement<VisitableElement<?>> visitab
     builder.endTerminalType();
   }
 
+  private static String asString(Serializable val) {
+    return val == null ? null : val.toString();
+  }
+
+  private static Number parseNumber(String str) {
+    NumberFormat numberFormat = NumberFormat.getInstance();
+    try {
+      return numberFormat.parse(str);
+    } catch (ParseException e) {
+      // Default to zero since this is called when reading FROM the database
+      // so fail gracefully because it must be a data persistence issue
+      return 0;
```

Comment:
✏️ Not sure how I feel about defaulting to 0, but I think we should at least log something in this case

Reply:
Fair - throwing an exception isn't really an option here unless I rewrite how form fetches work. Adding logging.

Reply:
Nvm I wrote form fetching correctly the first time there are just holes because I didn't want to "over-catch" runtime exceptions back in the day. Not sure what I was thinking at the time but it seems reasonable to be extra cautious and not allow **any** case where an exception mid-transform can fail the entire GET request. 

I've added a commit updating things accordingly. Now, throwing an exception here is definitely an option but I'm not sure it's the best option. At least with a poor default the user can still use their form and just change out the value on the fly.